File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.py
from collections import deque
def solution(stones, k):
    answer = 0
    
    lt = 1
    rt = max(stones)+1
    
    while lt<rt-1:
        mid = (lt+rt)//2
        work = 0
        flag = True
        for stone in stones:
            if stone<mid:
                work +=1
            else:
                work = 0
            if work ==k:
                flag= False
                break
        if flag:
            lt = mid
        else:
            rt = mid
    
    
    return lt

File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.java
import java.util.*;
class Solution {
    public int solution(int[] stones, int k) {
        int answer = 0;
        int lt = 0;
        int rt = Arrays.stream(stones).max().getAsInt()+1;
        
        while(lt<rt-1){
            int mid = (lt+rt)/2;
            
            boolean flag = true;
            int go = 0;
            for(int stone: stones){
                if(stone < mid){
                    go++;
                }else if(stone >= mid){
                    go = 0;
                }
                if(go == k){
                    flag = false;
                    break;
                }
            }
            if(!flag){
                rt = mid ;
            }else{
                lt = mid;
            }
        }
        
        
        return lt;
    }
}